lib/clients/incoming-webhook/client.js

```diff
@@ -0,0 +1,81 @@
+var requestsTransport = require('../transports/request').requestTransport;
+var isString = require('lodash').isString;
+var isObject = require('lodash').isObject;
+var noop = require('lodash').noop;
+
+/**
+ *
+ * @param {String} slackUrl
+ * @param {object?} defaults
+ * @param {string} defaults.username The default username to use when sending a webhook.
+ *      If no username is specified, the one chosen when creating the webhook will be used.
+ * @param {string} defaults.iconEmoji The default emoji to use when sending a webhook.
+ *      If no iconEmoji is specified, the one chosen when creating the webhook will be used.
+ * @param {string} defaults.channel The default channel to use when sending a webhook.
+ *      If no channel is specified, the one chosen when creating the webhook will be used.
+ * @param {string} defaults.text The default text to use when sending a webhook.
+ * @constructor
+ */
+function IncomingWebhook(slackUrl, defaults) {
+  var _defaults = defaults || {};
+
+  if (!slackUrl) {
+    throw new Error('Slack url is required');
+  }
+
+  this._slackUrl = slackUrl;
+  this.defaults = {
+    username: _defaults.username,
+    iconEmoji: _defaults.iconEmoji,
+    channel: _defaults.channel,
+    text: _defaults.text
+  };
+
+  this._transport = _defaults._transport || requestsTransport;
+}
+
+
+/**
+ * Sends a message via an incoming webhook
+ * @param {String|Object} message The message to send. Can be text or an object that
+ *     overrides the defaults in initialization.
+ * @param {Function=} optCb
+ */
+IncomingWebhook.prototype.send = function send(message, cb) {
+  var data = this._formatData(message);
+  var _cb = cb || noop;
+
+  this._transport({
+    url: this._slackUrl,
+    body: data
+  }, _cb);
+};
+
+
+/**
+ * Formats the data to be delivered to the transport
+ * @private
+ */
+IncomingWebhook.prototype._formatData = function _formatData(message) {
+  var data = {
+    username: this.defaults.username,
+    icon_emoji: this.defaults.iconEmoji,
```

Comment:
I would be of the mind to require the `:emoji:` form, and then do some checking to make sure what we got passed is of that form. We can add that later ;)